src/celeritas/optical/detail/ScintPreGenAction.cu

```diff
@@ -30,13 +30,13 @@ namespace detail
 void ScintPreGenAction::pre_generate(CoreParams const& core_params,
                                      CoreStateDevice& core_state) const
 {
+    auto& state
+        = get<OpticalGenState<MemSpace::native>>(core_state.aux(), data_id_);
+
     TrackExecutor execute{core_params.ptr<MemSpace::native>(),
                           core_state.ptr(),
                           detail::ScintPreGenExecutor{
-                              scintillation_->device_ref(),
-                              storage_->obj.state<MemSpace::native>(
-                                  core_state.stream_id(), core_state.size()),
-                              storage_->size[core_state.stream_id().get()]}};
+                              scintillation_->device_ref(), state.store.ref()}};
```

Comment:
Is this missing the buffer size?
```suggestion
                              scintillation_->device_ref(), state.store.ref(), state.buffer_size}};
```

Reply:
Goodness, why did that compile, and even worse how does the test pass? 👀 Good catch. I've checked the other call signatures.